src/domain/thirdpartyRequests/authorizations.ts

```diff
@@ -0,0 +1,176 @@
+    )
+    Logger.info(`authorications::forwardPostAuthorization - Resolved destination party ${endpointType} endpoint for thirdpartyTransaction: ${transactionRequestId} to: ${inspect(endpoint)}`)
+    const url: string = Mustache.render(endpoint + path, { ID: transactionRequestId })
+    Logger.info(`authorications::forwardPostAuthorization - Forwarding authorization to endpoint: ${url}`)
+
+    await Util.Request.sendRequest(
+      url,
+      headers,
+      sourceDfspId,
+      destinationDfspId,
+      Enum.Http.RestMethods.POST,
+      payload,
+      Enum.Http.ResponseTypes.JSON,
+      childSpan
+    )
+
+    Logger.info(`authorications::forwardPostAuthorization - Forwarded thirdpartyTransaction authorization: ${transactionRequestId} from ${sourceDfspId} to ${destinationDfspId}`)
+    if (childSpan && !childSpan.isFinished) {
+      childSpan.finish()
+    }
+  } catch (err) {
+    Logger.error(`authorications::forwardPostAuthorization - Error forwarding thirdpartyTransaction authorization to endpoint: ${inspect(err)}`)
+    const errorHeaders = {
+      ...headers,
+      'fspiop-source': Enum.Http.Headers.FSPIOP.SWITCH.value,
+      'fspiop-destination': sourceDfspId
+    }
+    const fspiopError: FSPIOPError = ReformatFSPIOPError(err)
+    await forwardPostAuthorizationError(
+      Enum.EndPoints.FspEndpointTemplates.THIRDPARTY_TRANSACTION_REQUEST_AUTHORIZATIONS_PUT_ERROR,
+      errorHeaders,
+      transactionRequestId,
+      fspiopError.toApiErrorObject(Config.ERROR_HANDLING.includeCauseExtension, Config.ERROR_HANDLING.truncateExtensions),
+      childSpan
+    )
+
+    if (childSpan && !childSpan.isFinished) {
+      await finishChildSpan(fspiopError, childSpan)
+    }
+    throw fspiopError
+  }
+
+}
+
+/**
+ * @function forwardPostAuthorizationError
+ * @description Generic function to handle sending `PUT .../authorizations/error` back to
+ *   the FSPIOP-Source
+ * @param {string} path Callback endpoint path
+ * @param {HapiUtil.Dictionary<string>} headers Headers object of the request
+ * @param {string} transactionRequestId the ID of the thirdpartyRequests/transactions resource
+ * @param {object} payload Body of the POST request
+ * @param {object} span optional request span
+ * @throws {FSPIOPError} Will throw an error if no endpoint to forward the transactions requests is
+ *  found, if there are network errors or if there is a bad response
+ * @returns {Promise<void>}
+ */
+export async function forwardPostAuthorizationError(path: string,
+  headers: HapiUtil.Dictionary<string>,
+  transactionRequestId: string,
+  payload: PostAuthorizationPayload,
+  span?: any): Promise<void> {
+  const childSpan = span?.getChild('forwardPostAuthorizationError')
+  const sourceDfspId = headers[Enum.Http.Headers.FSPIOP.SOURCE]
+  const destinationDfspId = headers[Enum.Http.Headers.FSPIOP.DESTINATION]
+  const endpointType = Enum.EndPoints.FspEndpointTypes.THIRDPARTY_CALLBACK_URL_TRANSACTION_REQUEST_AUTHORIZATIONS_PUT_ERROR
+
+  try {
```

Comment:
getEndpoint and Mustache.render are always in pair - maybe we can abstract away this pattern?

Reply:
Yeah maybe - I opened [this ticket](https://app.zenhub.com/workspaces/project-poc-59edee71d1407922110cf083/issues/mojaloop/project/1573) to follow up the conversation if we want to do this.